lib/phoenix/channel.ex

```diff
@@ -137,14 +137,18 @@ defmodule Phoenix.Channel do
                                                                   :ignore |
                                                                   {:error, reason :: term, Socket.t}
 
-  defcallback leave(msg :: map, Socket.t) :: {:ok, Socket.t}
+  defcallback leave(msg :: map, Socket.t) :: :ok | {:error, reason :: term}
```

Comment:
What was our conclusion regarding `leave`? I think we were discussing to rename it to terminate, weren't we? I don't remember the rationale anymore.